hazelcast/src/main/java/com/hazelcast/scheduledexecutor/impl/ScheduledExecutorContainer.java

```diff
@@ -61,24 +62,28 @@
 
     private final int durability;
 
+    private final int capacity;
+
     ScheduledExecutorContainer(String name, int partitionId, NodeEngine nodeEngine,
-                                      int durability) {
-        this(name, partitionId, nodeEngine, durability, new ConcurrentHashMap<String, ScheduledTaskDescriptor>());
+                                      int durability, int capacity) {
+        this(name, partitionId, nodeEngine, durability, capacity, new ConcurrentHashMap<String, ScheduledTaskDescriptor>());
     }
 
     ScheduledExecutorContainer(String name, int partitionId, NodeEngine nodeEngine,
-                                      int durability, ConcurrentMap<String, ScheduledTaskDescriptor> tasks) {
+                                      int durability, int capacity, ConcurrentMap<String, ScheduledTaskDescriptor> tasks) {
         this.logger = nodeEngine.getLogger(getClass());
         this.name = name;
         this.nodeEngine = nodeEngine;
         this.executionService = (InternalExecutionService) nodeEngine.getExecutionService();
         this.partitionId = partitionId;
         this.durability = durability;
+        this.capacity = capacity;
         this.tasks = tasks;
     }
 
     public ScheduledFuture schedule(TaskDefinition definition) {
         checkNotDuplicateTask(definition.getName());
+        checkNotAtCapacity();
```

Comment:
shouldn't we add this check for member owned container too? it overrides the `schedule` method